packages/api-runner/lib/fetchers/getPoolChainData.js

```diff
@@ -89,21 +91,11 @@ export const getPoolChainData = async (chainId, poolGraphData, fetch) => {
         )
     )
 
-    // Token Listener
-    // NOTE: If it's not a token faucet, this will break everything
-    if (pool.tokenListener.address) {
-      const tokenFaucetContract = contract(
-        pool.tokenListener.address,
-        TokenFaucetABI,
-        pool.tokenListener.address
-      )
-      batchCalls.push(
-        tokenFaucetContract
-          .dripRatePerSecond()
-          .asset()
-          .measure()
-      )
-    }
+    // Token Faucets
+    pool.tokenFaucets.forEach((tokenFaucetAddress) => {
```

Comment:
```suggestion
    pool.tokenFaucets?.forEach((tokenFaucetAddress) => {
```

So we're not dependent on adding the empty array to the `current-pool-data`